sklearn/utils/extmath.py

```diff
@@ -274,6 +274,7 @@ def randomized_range_finder(
            [-0.82..., -0.38...]])
     """
     xp, is_array_api_compliant = get_namespace(A)
+    A = check_array(A, accept_sparse=True)
```

Comment:
Similar comment as for `_randomized_svd`: we need to introduce a private `_randomized_range_finder` to remove the overhead of multiple input checks.